client/my-sites/stats/features/modules/stats-utm/stats-module-utm.jsx

```diff
@@ -47,8 +50,32 @@ const StatsModuleUTM = ( {
 	const siteId = useSelector( getSelectedSiteId );
 	const siteSlug = useSelector( ( state ) => getSiteSlug( state, siteId ) );
 	const translate = useTranslate();
-
+	const isOdysseyStats = config.isEnabled( 'is_running_in_jetpack_site' );
 	const [ selectedOption, setSelectedOption ] = useState( OPTION_KEYS.SOURCE_MEDIUM );
+	const [ url ] = useState( () => {
+		let initialUrl = summaryUrl;
+
+		if ( ! initialUrl ) {
+			initialUrl = isOdysseyStats ? window.location.hash.split( '#!' )[ 1 ] : window.location.href;
```

Comment:
A unified way to deal with Odyssey Stats parameters is to read params from context.

Reply:
[For example](https://github.com/Automattic/wp-calypso/blob/822d69390fc9ea28b33998f567d3ad40b939e42d/client/my-sites/stats/site.jsx#L264-L274), we can read query parameter and apply them to the module.